dev_tools/continous_integration_macos.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+# This script should only be run by Travis
+if [ "$TRAVIS_OS_NAME" == "osx" ]; then
+
+    # Install some custom requirements on macOS
+    # e.g. brew install pyenv-virtualenv
+    PYTHON=3.7.0
+    brew update
+    brew install openssl readline pyenv-virtualenv
+    brew outdated pyenv || brew upgrade pyenv
+    pyenv install $PYTHON
+    export PYENV_VERSION=$PYTHON
+    export PATH="/Users/travis/.pyenv/shims:${PATH}"
```

Comment:
Can the start of path we are adding be obtained e.g. by using `pwd`? Right now it is *very* specific to the setup travis-ci uses.